valohai/distributed.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import List, Optional
+
+from valohai import paths
+from valohai.internals.distributed_config import DistributedConfig, Member
+from valohai.internals.distributed_config.consts import MASTER_MEMBER_ID
+
+
+class Distributed:
+    """
+    Distributed toolkit accessed through `valohai.distributed`.
+    """
+
+    _config: Optional[DistributedConfig] = None
+
+    def is_distributed_task(self) -> bool:
+        # not a property to mimic `is_running_in_valohai`
+        try:
+            return bool(self.config.group_name)
+        except FileNotFoundError:
+            return False
+        except json.decoder.JSONDecodeError:
+            return False
+
+    @property
+    def group_name(self) -> str:
+        return self.config.group_name
+
+    @property
+    def member_id(self) -> str:
+        return self.config.member_id
+
+    @property
+    def rank(self) -> Optional[int]:
+        return self.self().rank
+
+    @property
+    def required_count(self) -> int:
+        return self.config.required_count
+
+    def members(self) -> List[Member]:
+        return self.config.members
+
+    def member(self, member_id: str) -> Member:
+        for member in self.members():
+            if member.member_id == member_id:
+                return member
+        raise RuntimeError(f"No member with id {member_id}")
+
+    def self(self) -> Member:
```

Comment:
`self.self()` seems so _wrong_, but the only other sane wording I can think of is `.me()`, which isn't better at all really.

Just a comment...

Reply:
yeah, I was thinking the same and we don't __have to__ go with the JSON key naming :thinking: 

will revisit